v2/api/eventhub/customizations/namespace_extension.go

```diff
@@ -0,0 +1,115 @@
+)
+
+var _ genruntime.KubernetesExporter = &NamespaceExtension{}
+
+func (ext *NamespaceExtension) ExportKubernetesResources(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+) ([]client.Object, error) {
+	// This has to be the current hub storage version. It will need to be updated
+	// if the hub storage version changes.
+	typedObj, ok := obj.(*storage.Namespace)
+	if !ok {
+		return nil, errors.Errorf("cannot run on unknown resource type %T, expected *eventhub.Namespace", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not
+	var _ conversion.Hub = typedObj
+
+	hasSecrets := namespaceSecretsSpecified(typedObj)
+	if !hasSecrets {
+		log.V(Debug).Info("No secrets retrieval to perform as operatorSpec is empty")
+		return nil, nil
+	}
+
+	id, err := genruntime.GetAndParseResourceID(typedObj)
+	if err != nil {
+		return nil, err
+	}
+
+	// Only bother calling ListKeys if there are secrets to retrieve
+	var res armeventhub.NamespacesClientListKeysResponse
+	if hasSecrets {
+		subscription := id.SubscriptionID
+		// Using armClient.ClientOptions() here ensures we share the same HTTP connection, so this is not opening a new
+		// connection each time through
+		var confClient *armeventhub.NamespacesClient
+		confClient, err = armeventhub.NewNamespacesClient(subscription, armClient.Creds(), armClient.ClientOptions())
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create new NamespaceClient")
+		}
+
+		// RootManageSharedAccessKey is the default auth rule for namespace.
+		// See https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-get-connection-string
+		res, err = confClient.ListKeys(ctx, id.ResourceGroupName, typedObj.AzureName(), "RootManageSharedAccessKey", nil)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to retreive response")
+		}
+
+	}
+
+	secretSlice, err := namespaceSecretsToWrite(typedObj, res.AccessKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	return secrets.SliceToClientObjectSlice(secretSlice), nil
+}
+
+func namespaceSecretsSpecified(obj *storage.Namespace) bool {
+	if obj.Spec.OperatorSpec == nil || obj.Spec.OperatorSpec.Secrets == nil {
+		return false
+	}
+
+	secrets := obj.Spec.OperatorSpec.Secrets
+
+	if secrets.PrimaryKey != nil ||
+		secrets.SecondaryKey != nil ||
+		secrets.PrimaryConnectionString != nil ||
+		secrets.SecondaryConnectionString != nil {
+		return true
+	}
+
+	return false
+}
+
+func namespaceSecretsToWrite(obj *storage.Namespace, keys armeventhub.AccessKeys) ([]*v1.Secret, error) {
+	operatorSpecSecrets := obj.Spec.OperatorSpec.Secrets
+	if operatorSpecSecrets == nil {
+		return nil, errors.Errorf("unexpected nil operatorspec")
+	}
+
+	collector := secrets.NewCollector(obj.Namespace)
+
+	collector.AddValue(operatorSpecSecrets.PrimaryKey, *keys.PrimaryKey)
```

Comment:
use `to.Value` rather than direct deref here to hedge against panics.

```suggestion
	collector.AddValue(operatorSpecSecrets.PrimaryKey, to.Value(keys.PrimaryKey))
```

Reply:
Same below